### PR TITLE
Feat: Enhance UseWebNotificationOptions and ShowOptions with additional field

### PR DESCRIPTION
**Pull request by hcandelaria**
- Added `vibration` field to allow specifying vibration patterns for notifications.

### CODE REVIEW

File: src/use_web_notification.rs [hunk 1]
```diff
@@ -277,10 +276,14 @@ pub struct UseWebNotificationOptions {
...
 
     /// A boolean value specifying whether the notification should be silent, regardless of the device settings.
-    /// The default is `false`, which means the notification is not silent. If `true`, then the notification will be silent.
+    /// The default is `null`, which means the notification is not silent. If `true`, then the notification will be silent.
     #[builder(into)]
     silent: Option<bool>,
 
+    /// A JsValue array specifying the vibration pattern in which the device is vibrating and not vibrating.
+    #[builder(into)]
+    vibrate: Option<wasm_bindgen::JsValue>,
```
Comment (maccesch) [inline src/use_web_notification.rs:285]: To make calling this SSR safe this shouldn't be a `JsValue`. Can we convert this a meaningfull Rust type?

Comment (hcandelaria) [reply src/use_web_notification.rs]: Thanks for the feedback!
Updated it to use `Vec<u16>` instead and added a helper function to convert to `JsValue`. Let me know if more changes are needed.

I also pulled the latest changes and applied formatting to `watch_debounced` and `watch_throttled`. Not sure if I should've done it this way, if you have any feedback let me know.

Comment (maccesch) [reply src/use_web_notification.rs]: Thanks! Could you quickly describe in the docs for that options what the numbers in the Vec mean exactly?

**Comment by maccesch** [comment]
There are now many files listed as shown as well as conflicts.

**Comment by hcandelaria** [comment]
Yeah sorry, it probably wasn't the best idea to pull the changes from main. I will work on your feedback to update the docs as well as the conflicts.

Thank you for all your help.

**Comment by hcandelaria** [comment]
It looks like the check failed because of the examples referencing `codee` version 0.1. Should I go ahead and update the version on those examples?

Sorry!😅 I have a bunch of commits on this branch, which probably does not make it any easier to review. Thank you for your feedback.

**Comment by maccesch** [comment]
Thanks for all you good work!

No worries about the codee version. I'll update that.

**Pull request merged**